### ISSUE TITLE
Noninteractive Basic Authentication Restore

### ISSUE THREAD

**Issue by jspaetzel** [NuGet/Home#2022]
I'm attempting to use nuget with a private nuget repository setup with ProGet using basic authentication to access packages. Interactive basic authentication works fine, im only having trouble with doing this non-interactively with my ci or directly via the command line.

I saw several other issues related to this but did not see one that is currently open. I've tried this on nuget.exe versions 3.3.1, 3.3.0, and 2.8.6. 

**This is what I am doing:**

```
nuget setapikey KEY -source URL
nuget sources update -Name SOURCENAME -source SOURCEURL -User USER -pass PASSWORD -StorePasswordInClearText

nuget restore -noninteractive
Installing 'PACKAGE'.
Please provide credentials for: ****
UserName: Cannot prompt for input in non-interactive mode.
```

Steps taken so far & other notes: 
- Confirmed WWW-Authenticate header is being sent back by proget via http intercept
- nuget sources correctly applied to my \AppData\Roaming\NuGet\NuGet.Config, have also tried using configs in other locations such as program data and specifying by -ConfigFile
- Using IIS with ProGet
- Using Windows 2012 server as well as a Windows 10 machine

My workaround for the moment is to have a local package cache on my build runners that I populate separately, but I would much rather use nuget to retrieve updated packages.

**Comment by yishaigalatzer** [comment]
Are you sure you are using the same user name on your CI machine?

**Comment by jspaetzel** [comment]
@yishaigalatzer I've tried using a different one that's specifically for that machine and also the same user. Either set works fine when I put them in interactively in the command line or from visual studio

**Comment by yishaigalatzer** [comment]
Can you share your nuget.config and can you try placing it at the root of your solution?

**Comment by jspaetzel** [comment]
@yishaigalatzer Just tried it at the root of my solution, no change. Here is essentially my current nuget.config though with credentials stripped of course. 

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageRestore>
    <add key="enabled" value="True" />
    <add key="automatic" value="True" />
  </packageRestore>
  <apikeys>
    <add key="xxxx" value="xxxx" />
  </apikeys>
  <packageSources>
    <add key="nuget.org" value="https://www.nuget.org/api/v2/" />
    <add key="MyNuget" value="xxxx" />
  </packageSources>
  <packageSourceCredentials>
    <MyNuget>
      <add key="Username" value="xxxx" />
      <add key="ClearTextPassword" value="xxxx" />
    </MyNuget>
  </packageSourceCredentials>
  <activePackageSource>
    <add key="All" value="(Aggregate source)" />
  </activePackageSource>
</configuration>
```

**Comment by jspaetzel** [comment]
I've tried these locations so far with that config file,
- project\NuGet.config
- project.nuget\NuGet.config
- C:\Users\User\AppData\Roaming\NuGet\NuGet.config
- C:\ProgramData\NuGet\NuGet.config

**Comment by yishaigalatzer** [comment]
@alpaix you can set up a proget server for free from their site to validate this scenario

**Comment by martinsuchan** [comment]
Same problem here, we cannot build our Windows 10 project on our build server right now:  

```
[17:55:36][restore] Zadejte pýihlaçovacˇ Łdaje pro: https://----.azurewebsites.net/nuget/
[17:55:36][restore] U§ivatelsk‚ jm‚no: Zadejte pýihlaçovacˇ Łdaje pro: https://----.azurewebsites.net/nuget/
[17:55:36][restore] U§ivatelsk‚ jm‚no: Warning: FindPackagesById: SznAuthorize
[17:55:36][restore]   V neinteraktivnˇm re§imu nelze zobrazit vězvu pro zad nˇ vstupu.
[17:55:36][restore] Warning: FindPackagesById: SznAuthorize
[17:55:36][restore]   V neinteraktivnˇm re§imu nelze zobrazit vězvu pro zad nˇ vstupu.
[17:55:36][restore] Zadejte pýihlaçovacˇ Łdaje pro: https://----.azurewebsites.net/nuget/
[17:55:36][restore] U§ivatelsk‚ jm‚no: Zadejte pýihlaçovacˇ Łdaje pro: https://----.azurewebsites.net/nuget/
[17:55:36][restore] Failed to retrieve information from remote source 'https://----.azurewebsites.net/nuget/'.
```

Our setup: Windows 10 Enterprise x64, TeamCity 9.1.6, Visual Studio 2015 Update 1, NuGet 3.3

**Comment by yishaigalatzer** [comment]
We have made quite a few changes in auth and proget support in 3.4, we are going to get to this issue in a few days, but in the meantime feel free to try the (almost RTM) build from - https://www.myget.org/F/nugetbuild/api/v2/package/NuGet.CommandLine/3.4.0-rtm-768

**Comment by alpaix** [comment]
@zzbomb I couldn't repro the scenario. Based on log messages it looks like a configuration issue when NuGet does not "see" the credentials or cannot associate them with a feed. Also APIKey is not relevant in restore command. 

Could you please try [3.4.0](https://dist.nuget.org/win-x86-commandline/latest-prerelease/nuget.exe)? If it still happens run it with `-Verbosity detailed` to troubleshoot the issue.

**Comment by jspaetzel** [comment]
@alpaix @yishaigalatzer This appears to be resolved in the 3.5.0.1053 build. Thanks!
